openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -2728,6 +2749,30 @@ struct AMDGPUDeviceTy : public GenericDeviceTy, AMDGenericDeviceTy {
     return Err;
   }
 
+  /// Detect if current architecture is an APU.
+  Error checkIfAPU() {
+    std::string StrGfxName(ComputeUnitKind);
+    std::transform(std::begin(StrGfxName), std::end(StrGfxName),
+                   std::begin(StrGfxName),
+                   [](char c) { return std::tolower(c); });
+    if (StrGfxName == "gfx940") {
+      IsAPU = true;
```

Comment:
Can we avoid depending on listing device names?

Reply:
What other ways do we have to detect an APU? It is either a gfx940 or a gfx942 with a specific chip ID. I don't see any other way to determine this.

Reply:
Really wish HSA exported some stable ID to use rather than strings, like the ones in the ELF headers.